tests/baselines/reference/numericLiteralTypes3.types

```diff
@@ -85,9 +85,9 @@ function f3(a: A, b: B, c: C, d: D) {
 >b : B
 
     c = c;
->c = c : B
+>c = c : 2 | 3

```

Comment:
This is an interesting reversion; any comments?

Reply:
It seems like, possibly, this happens because when you narrow, you're going to construct a new union. Previously, you'd have an existing type ID to match to an alias.

I think the weird thing to me is that even if C was declared in terms of B, you wouldn't see it. I think in that case you'd need to introduce origin type narrowing. Not able to test right now though.

Reply:
Seems like an interesting side effect of control flow. In the preceding line, `c: C` was narrowed to `c: B`, but reassigning `c: B` to itself seems to change the type?

Reply:
The `c = b` assignment on the preceding line narrows the type of `c` to those constituents of `c` for which `b` might provide a value. In other words, `2 | 3`. This happens to be the same type as `B`, but it is strictly speaking unrelated. We could add logic to recognize that it exactly matches the assigned type and then go with the assigned type which would pick up the alias. Not sure it's that important though.